packages/contracts-bridge/contracts/BridgeMessage.sol

```diff
@@ -388,6 +393,26 @@ library BridgeMessage {
         return _transferAction.indexAddress(13);
     }
 
+    /**
+     * @notice Retrieves the sender from a TransferWithHook
+     * @param _transferAction The message
+     * @return The sender as bytes32
+     */
+    function sender(bytes29 _transferAction)
+        internal
+        pure
+        typeAssert(_transferAction, Types.TransferToHook)
+        returns (bytes32)
+    {
+        // before = 1 byte identifier + 32 bytes hook address + 32 bytes amount + 32 bytes detailsHash = 97
+        return _transferAction.index(97, 32);
+    }
+
+    /**
+     * @notice Retrieves the extra data from a TransferWithHook
+     * @param _transferAction The message
+     * @return The extra data as a bytes29
```

Comment:
nit: a typed pointer to the extra data in memory

Reply:
https://github.com/nomad-xyz/monorepo/pull/414/commits/75b3d2d83f8c8fce1c06471a2fcdee2c488f9dee